feat: docs to update ENV values in helm installation (#707)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/instance-configuration/README.mdx

---
description: >-
  Appsmith ships with third-party services & configurations that improve the app
  building experience. All services & configurations are entirely optional.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Instance Configuration

Configuration of your Appsmith instance allows you to tailor it to your specific needs. This process includes setting up elements like environment variables, authentication, emails, SSL, and custom domain. Proper configuration ensures that your instance is running efficiently. 

## Configure Docker installations

To configure a docker installation, go to your installation folder and edit the `stacks/configuration/docker.env` file with the environment variables for the service.

> Remove `#` before the variables to ensure they're not commented.

Reviewer: @sharat87 - Do we need this line? Can't we add parameters without `#`?

Reviewer 2: Not strictly. Can be removed.

---
description: >-
  Appsmith ships with third-party services & configurations that improve the app
  building experience. All services & configurations are entirely optional.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Instance Configuration

Configuration of your Appsmith instance allows you to tailor it to your specific needs. This process includes setting up elements like environment variables, authentication, emails, SSL, and custom domain. Proper configuration ensures that your instance is running efficiently. 

## Configure Docker installations

> To configure a docker installation, simply open the folder of your installation and edit the **`stacks/configuration/docker.env`** file with the environment variables for the service.

Reviewer 2: ```suggestion
To configure a docker installation, go to your installation folder and edit the `stacks/configuration/docker.env` file with the environment variables for the service.
```

> Remove # before the variables to ensure they're not commented

Reviewer 2: ```suggestion
Remove `#` before the variables to ensure they're not commented.
```

```bash

> // Example variable configuration format

Reviewer 2: Can we remove these comments from the code blocks. On the rendered documentation, these code blocks show up with a `copy` button. It is expected that people can just hit that button to copy the code, and paste in their terminal.

Having this comment _inside_ the code block means that they also copy this comment. So, can we either move this comment out into the previous sentence, or just remove it please?

APPSMITH_GOOGLE_MAPS_API_KEY=YOUR_API_KEY
```

After making any changes, **remember to restart the docker containers** for the changes to take effect.

```bash
// To restart Appsmith using docker
docker restart appsmith

// To restart Appsmith using docker compose
docker-compose restart appsmith

> ```

Reviewer 2: Similar problem as above. Let's remove comments, _and_ split this into two code blocks, with the comment as just ordinary text, so it's easy to copy just any one command.

> ## Configure Kubernetes installations
> 
> To configure a `k8s` installation, simply open the folder of your installation and edit the `config-template/appsmith-configmap.yaml` file.
> 
> Remove # before the variables to ensure they're not commented
> 
> ```
> // Example variable configuration format
> APPSMITH_GOOGLE_MAPS_API_KEY: "YOUR_API_KEY"
> ```
> 
> After making any changes, **remember to restart the pods** for the changes to take effect
> 
> ```
> // commands to restart k8s pods
> kubectl apply -f appsmith-configmap.yaml
> kubectl scale deployment appsmith-internal-server --replicas=0
> kubectl scale deployment appsmith-internal-server --replicas=1
> ```
> 

Reviewer 2: Yes, let's remove this.

```suggestion
```

```

## Configure Helm installations

To configure environment variables using helm, follow these steps:

<Tabs groupId="appsmithEditions">
  <TabItem label="Community Edition" value="communityEdition"> 

  You may choose to configure environment variables: through commands or by using a `values.yaml` file.

  * When using command, specify each parameter using `--set key=value[,key=value]` argument to helm upgrade.

```bash
  helm upgrade --set applicationConfig.<ENV_KEY>=<ENV_VALUE> [ RELEASE ] [ CHART ]  
```

>   For example, the below command sets the `APPSMITH_SIGNUP_DISABLED` variable to `true` and configures it to disable signup button on Appsmith login screen. 

Reviewer 2: Ensure your code blocks and content line up to the bullet point's content, so they all show up aligned in the rendered website. In this case, we currently have this:

![Screenshot 2022-12-29 at 10 14 13 AM](https://user-images.githubusercontent.com/120119/209904722-bf50f4ef-8924-4d3b-9d08-e9e48b04779d.png)

If we align to the content of the bullet point, which is `When using...`:
![Screenshot 2022-12-29 at 10 15 40 AM](https://user-images.githubusercontent.com/120119/209904581-ceab35a6-faf4-4412-9897-aaca0c8e6ecd.png)

Then we get this:

![Screenshot 2022-12-29 at 10 16 51 AM](https://user-images.githubusercontent.com/120119/209904730-6a530604-7125-4c37-bcc8-2825e90043a6.png)

See the difference? Can we fix this for all the other code blocks under bullet points as well please?

Reviewer: Got it.

```bash
  helm upgrade --values values.yaml appsmith appsmith/appsmith
```

  </TabItem>
  <TabItem label="Business Edition" value="businessEdition"> 

   You may choose to configure environment variables: through commands or by using a `values.yaml` file.

  * When using command, specify each parameter using `--set key=value[,key=value]` argument to helm upgrade.

```bash
  helm upgrade --set applicationConfig.<ENV_KEY>=<ENV_VALUE> [ RELEASE ] [ CHART ]  
```
  For example, the below command sets the `APPSMITH_SIGNUP_DISABLED` variable to `true` and configures it to disable signup button on Appsmith login screen. 

  * Alternatively, use a `values.yaml` file. You can specify the parameter values in the `applicationConfig` section. 

```yaml
  APPSMITH_OAUTH2_GOOGLE_CLIENT_ID: ""
  APPSMITH_OAUTH2_GOOGLE_CLIENT_SECRET: ""
  APPSMITH_OAUTH2_GITHUB_CLIENT_ID: ""
  APPSMITH_OAUTH2_GITHUB_CLIENT_SECRET: ""
  APPSMITH_FORM_LOGIN_DISABLED: ""
  APPSMITH_SIGNUP_DISABLED: "true"
```
Once you have made changes to the values file, run the below command.

```bash
  helm upgrade --values values.yaml appsmith appsmith-ee/appsmith
```
  </TabItem>
</Tabs> 

> ## Configure ECS installations

Reviewer 2: This section has too much bold text. We've used bold text in above sections, but only for text we absolutely don't want our users to miss, and grab their attention towards it.

But in this section we seem to be using it for... several things. Emphasis on too many things, gives emphasis to nothing. Can we reduce the bold here please?

Reviewer: Which parts need emphasis? I can make the changes.

Reviewer: @sharat87 - I can do the above minor changes as part of this PR. On another note, we'll have to rewrite this page as for now it just focuses on Configuring Environment variables. However, it's more of a landing page that will give users an idea on:
- Introduction for Instance Configuration
- Things to be expected like custom domain, email config and more
- Configuring Environment variables for different installation types
- Troubleshooting section
- Further reading.
Do let me know your thoughts? I'll raise another ticket to look into this.

Reviewer 2: Oh hey, sorry I thought this was all written by Sumesh, saw his name on the PR, so my comments were targeted towards him, didn't know you're working on this. 🙂

Reg. your question, honestly, I don't think anything in this section needs an emphasis. It's just fine being plain, but I'm not too strong on this. If you see something that should be bold, please go ahead. 🙂

Reviewer: Sumesh has only added the configured Helm installations section and passed it on to me for rewriting. The others are not yet worked on. But it makes sense to fix the things as part of this PR as are no huge changes. 
I'll get the changes done, and we can merge this PR. We can discuss if the page needs restructuring and if needed can do a separate PR to accommodate that.

To configure an ECS installation, follow these steps:

1. Navigate to the **ECS console** and select **Task Definitions** on the sidebar.
2. Click on the Task Definition used by your ECS instance, and hit **Create new revision**.
3. On the `Task Definition config` page, click the Appsmith **container definition** to edit it. In the **Environment Section**, enter the environment configuration as **key-value pairs** as shown below.

![](/img/ecs-task-env_(1).png)

> It's recommended to use **AWS Secrets** for sensitive information, please follow the steps to [create a secret](https://docs.aws.amazon.com/secretsmanager/latest/userguide/manage\_create-basic-secret.html). Use the **ARN** of the secret as the Environment value and set the option to **ValueFrom**.

Reviewer 2: I don't think we need to escape this anymore. We're not on Gitbook anymore, and Docusaurus' Markdown parser is quite smart.

```suggestion
It's recommended to use **AWS Secrets** for sensitive information, please follow the steps to [create a secret](https://docs.aws.amazon.com/secretsmanager/latest/userguide/manage_create-basic-secret.html). Use the **ARN** of the secret as the Environment value and set the option to **ValueFrom**.
```
